addons/project/models/project_project.py

```diff
@@ -416,6 +416,20 @@ def copy(self, default=None):
                 old_project.map_tasks(new_project.id)
             if not self.active:
                 new_project.with_context(active_test=False).tasks.active = True
+            # Copy the shared embedded actions in the new project
+            shared_embedded_actions = self.env['ir.embedded.actions'].search([
+                ('parent_res_id', '=', old_project.id),
+                ('parent_res_model', '=', 'project.project'),
+                ('user_id', '=', False),
+            ])
+            copy_shared_embedded_actions = shared_embedded_actions.copy({'parent_res_id': new_project.id})
+            for i in range(len(shared_embedded_actions)):
+                filter_ids = shared_embedded_actions[i].filter_ids
+                copy_filters = filter_ids.copy()
+                copy_filters.embedded_parent_res_id = new_project.id
+                for j in range(len(filter_ids)):
+                    copy_filters[j].name = filter_ids[j].name
```

Comment:
Rather than letting the name be changed then change it back, I think we could prevent it from being changed in the first place by adding a context key